Makefile

```diff
@@ -53,51 +65,18 @@ update_circleci_reference: test
 #build_32bit: build_environment
 #	COMPILED_TAG_NAME=32bit COMPILE_OPTION=32BIT=Y $(MAKE) build
 
-build_debug: build_environment
-	COMPILED_TAG_NAME=debug COMPILE_OPTION="REPRO=\\\nDEBUG=Y" $(MAKE) build
-
-# We might *have* serialized data (pulled from cloud) and not want to rebuild the model...
-build_environment_serialize: build_environment
-	if [ ! -d "$(SERIALBOX_PATH)" ];then \
-		git clone $(SERIALBOX_REPO) $(SERIALBOX_PATH);\
-	fi
-	ENVIRONMENT_TAG_NAME=serialize \
-	DOCKERFILE=docker/Dockerfile.serialize \
```

Comment:
It still exists as a build target in the docker file. `fv3ser` could invoke that file directly to create the base image. I use a similar style in the `fv3gfs-python` repo to call the docker image in the `fv3gfs-fortran` repo to create the fortran base images.

Reply:
If we end up building these images with specific targets in a bunch of different repos, I think it makes sense to have those build options be part of this repo..Similarly generating the serialized data -- it's a little odd to me to not have a target to do that in this repo, it sort renders. We talked about this and I had agreed to do it in fv3ser, but I still think it belongs in this repo.  I think we have been working with slightly different mental models of how this will be organized -- e.g. I think of fv3ser as the start to the new python model, that has tests with regression data generated from fv3gfs-fortran, while you see it as the serialization model run management repo... which is not a bad way to think about it, just different than what I was planning. And not a big deal -- it's a difference of fv3ser doing a 'docker run' vs a 'make run,' so maybe I shouldn't get too caught up on it.

Reply:
There were similar reasons why [`build_docker.sh`](https://github.com/VulcanClimateModeling/fv3gfs-python/blob/master/build_docker.sh) in fv3gfs-python builds targets from fv3gfs-fortran as a base for doing its tasks. `fv3gfs-fortran` on its own has no use for the intermediate images `fv3gfs-python` requires, so it doesn't make sense to add a make target for them. It isn't a problem because `fv3gfs-python` can build those targets explicitly.

However, if we decide to produce these intermediate images as GCR-pushed images, that might be a worthwhile reason to build the targets in this repo, and it could cut down compile time a lot (sometimes the docker cache throws away ESMF and FMS and they need to be rebuilt on CircleCI). The problem is I haven't thought of a good solution to version management in this context - do we version all of these sub-images? Where do those versions live? etc.

I think it's good to place the responsibility of building and providing a model that produces serialized output on this repo, and we've done that here. I think the responsibility of using that model to produce serialized output and testing it against gt4py stencils belongs in `fv3ser`.

In terms of the question right here, these two ideas ("fv3ser is the start of the new dycore" and "fv3ser manages serialization model runs") aren't incompatible - can't fv3ser do both? In my mind, the testing code for a product belongs where the product lives, and that's why fv3ser should contain all the pieces required to test the serialized components. I see fv3ser as containing the new dycore and its test code.

Personally I see `fv3ser`, the new dycore, `fv3gfs-python`, `fv3util`, `cyppy` (or some other physics treatment) and `fv3config` coming together to form a model when all the pieces are in place.

Reply:
I agree with most of this -- though  -- the test code should definitely live with the model, but regression data can be generated in any number of ways. It could be someone's Matlab script or in a database containing inputs and output data to some repo's functions. Being able to generate it from the same repo is of course also good form -- though once we have a model validated against the fortran, we should be able to generate regression data directly from the python and not need the fortran model anymore. Having it in fv3ser will make some things easier for sure on our way to getting there though. We can always change things later. I'll just be careful about how I set that up.